numba/stencilparfor.py

```diff
@@ -269,6 +269,69 @@ def _mk_stencil_parfor(self, label, in_args, out_arr, stencil_ir,
                                        self.calltypes)
             equiv_set.insert_equiv(out_arr, in_arr_dim_sizes)
             init_block.body.extend(stmts)
+        else: # out is present
+            if "cval" in stencil_func.options: # do out[:] = cval
+                cval = stencil_func.options["cval"]
+                # TODO: Loosen this restriction to adhere to casting rules.
+                if return_type.dtype != typing.typeof.typeof(cval):
+                    msg = "cval type does not match stencil return type."
+                    raise ValueError(msg)
+
+                # get slice ref
+                slice_var = ir.Var(scope, mk_unique_var("$py_g_var"), loc)
+                for fn, tmplt in numba.typing.templates.builtin_registry.globals:
+                    if fn == slice:
+                        slice_template = tmplt
+                        break
+                self.typemap[slice_var.name] = slice_template
```

Comment:
282-285 make a global or a function.  any weird possibility of not finding slice?

Reply:
> 282-285 make a global or a function.

Not sure what you mean? Do you mean to refactor this into a function in ir_utils.py called e.g. `get_slice_global` or to create a global/function in the generated IR for something?!

>  any weird possibility of not finding slice?

I suppose it's possible to break things so that `slice` might not be found, but I expect cases where this occurs will have greater problems.

Reply:
Ah, it's not actually storing the call-template.  The variable name is misleading.  It has the function type.

Reply:
That's a better method, thanks.